distribution/push_v2.go

```diff
@@ -169,7 +170,13 @@ func (p *v2Pusher) pushV2Tag(ctx context.Context, ref reference.NamedTagged, ima
 
 	putOptions := []distribution.ManifestServiceOption{distribution.WithTag(ref.Tag())}
 	if _, err = manSvc.Put(ctx, manifest, putOptions...); err != nil {
-		logrus.Warnf("failed to upload schema2 manifest: %v - falling back to schema1", err)
+		logrus.Warnf("failed to upload schema2 manifest: %v", err)
+
+		if runtime.GOOS == "windows" {
+			return err
+		}
+
+		logrus.Warn("Falling back to schema1")
```

Comment:
I'd prefer to keep this combined with the `failed to upload schema2 manifest` log message. If there are concurrent operations going on, the `Falling back to schema1` could show up later in the log, and it may not be obvious what operation it refers to.

It would be a good idea to also include the endpoint in this message so the user knows which registry was involved, but that's outside the scope of this PR.